main.py

```diff
@@ -195,24 +195,19 @@ def do_match(cls, flow, args):
             return False
         for name in flow['flow']['match']:
             field_flow = flow['flow']['match'][name]
+            field = args.get(name)
             if name == 'dl_vlan':
-                field = args.get(name)
-                if field:
-                    field = field[-1]
                 if not match_field_dl_vlan(field, field_flow):
                     return False
                 continue
-            if name not in args:
+            if not field:
```

Comment:
This condition is there because `name == dl_vlan` means `dl_vlan=0`. So, for that case, the absence of the field requires a different treatment. Right?

Reply:
Right. That's correct, even if the packet args doesn't have a `dl_vlan`, but if flow match keys have `dl_vlan` then indeed the match should be tried, so yes, we can't move this earlier, so, please add a further comment, on this `if not field` with a brief comment explaining just so future developers don't get confused why this isn't like a full early return.